prowler/providers/aws/services/iam/iam_service.py

```diff
@@ -99,44 +103,43 @@ def __get_credential_report__(self):
             credential_lines = credential.split("\n")
             csv_reader = csv.DictReader(credential_lines, delimiter=",")
             credential_list = list(csv_reader)
-            return credential_list
         except Exception as error:
             logger.error(
                 f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
-            return []
+        finally:
+            return credential_list
 
     def __get_groups__(self):
         try:
-            get_groups_paginator = self.client.get_paginator("list_groups")
-        except Exception as error:
-            logger.error(
-                f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
-            )
-        else:
             groups = []
+            get_groups_paginator = self.client.get_paginator("list_groups")
             for page in get_groups_paginator.paginate():
                 for group in page["Groups"]:
                     if not self.audit_resources or (
                         is_resource_filtered(group["Arn"], self.audit_resources)
                     ):
-                        groups.append(Group(group["GroupName"], group["Arn"]))
+                        groups.append(Group(name=group["GroupName"], arn=group["Arn"]))
 
+        except Exception as error:
+            logger.error(
+                f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+        finally:
             return groups
 
     def __get_account_summary__(self):
         try:
             account_summary = self.client.get_account_summary()
+            return account_summary
```

Comment:
What happens if the `self.client.get_account_summary()` raises an Exception?

Reply:
Changed